src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/problem/ExceptionHandling.kt

```diff
@@ -0,0 +1,26 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.problem
+
+import org.springframework.security.access.AccessDeniedException
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException
+import org.springframework.web.bind.annotation.ControllerAdvice
+import org.zalando.problem.StatusType
+import org.zalando.problem.ThrowableProblem
+import org.zalando.problem.spring.common.AdviceTrait
+import org.zalando.problem.spring.web.advice.ProblemHandling
+
+@ControllerAdvice
+class ExceptionHandling : ProblemHandling {
+  override fun toProblem(throwable: Throwable, status: StatusType): ThrowableProblem? {
+    if (throwable is AuthenticationCredentialsNotFoundException) {
+      return UnauthenticatedProblem()
+    }
+
+    if (throwable is AccessDeniedException) {
+      return ForbiddenProblem()
+    }
+
+    return AdviceTraitDefault().toProblem(throwable, status)
```

Comment:
Does this return `json+problem` responses? Interesting, I'll keep an eye on this.

Reply:
It does indeed, we're using https://github.com/zalando/problem-spring-web for this.  It works quite well out of the box but for some of the authentication failure conditions (e.g. expired JWT) I had to override a few bits - this class being part of that